.github/workflows/presubmit.yml

```diff
@@ -243,80 +125,47 @@ jobs:
-      - run: sudo apt update; sudo apt install -y $APT_PACKAGE
-      - run: sudo gem install cast-to-yaml nokogiri babeltrace2 opencl_ruby_ffi metababel
-      - name: Load Efficios Dependencies
+      - uses: ./.github/actions/setup-thapi
+      - name: Make distcheck
         run: |
-          echo "$HOME/efficios_dep/bin" >> $GITHUB_PATH
-          echo "PKG_CONFIG_PATH=$HOME/efficios_dep/lib/pkgconfig:$PKG_CONFIG_PATH" >> $GITHUB_ENV
-          echo "CPATH=$HOME/efficios_dep/include:$CPATH" >> $GITHUB_ENV
-          echo "LD_LIBRARY_PATH=$HOME/efficios_dep/lib:$LD_LIBRARY_PATH" >> $GITHUB_ENV
-          echo "LIBRARY_PATH=$HOME/efficios_dep/lib:$LIBRARY_PATH" >> $GITHUB_ENV
-      - run: mkdir -p build
-      - run: ./autogen.sh
-      - run: ../configure
-        working-directory: build
-      - run: make distcheck -j
-        working-directory: build
+          ./autogen.sh
+          mkdir -p build
+          cd build
+          ../configure
+          make -j distcheck
         env:
           THAPI_VALGRIND: 1
 
   dist-check:
-    needs: [efficios_dep, pre_job]
+    needs: [pre_job]
     if: ${{ needs.pre_job.outputs.should_skip != 'true' }}
     name: Dist and Check ubuntu-24.04
     runs-on: ubuntu-24.04
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/cache@v4
-        id: efficios_dep
-        env:
-          cache-name: cache-efficios_dep
-        with:
-          path: ~/efficios_dep/
-          key: ${{ runner.os }}-build-${{ env.cache-name }}
-      - run: sudo apt update; sudo apt install -y $APT_PACKAGE
-      - run: sudo gem install cast-to-yaml nokogiri babeltrace2 opencl_ruby_ffi metababel
-      - name: Load Efficios Dependencie
+      - uses: ./.github/actions/setup-thapi
+      - name: Make dist
         run: |
-          echo "$HOME/efficios_dep/bin" >> $GITHUB_PATH
-          echo "PKG_CONFIG_PATH=$HOME/efficios_dep/lib/pkgconfig:$PKG_CONFIG_PATH" >> $GITHUB_ENV
-          echo "CPATH=$HOME/efficios_dep/include:$CPATH" >> $GITHUB_ENV
-          echo "LD_LIBRARY_PATH=$HOME/efficios_dep/lib:$LD_LIBRARY_PATH" >> $GITHUB_ENV
-          echo "LIBRARY_PATH=$HOME/efficios_dep/lib:$LIBRARY_PATH" >> $GITHUB_ENV
-      - run: mkdir -p build
-      - run: ./autogen.sh
-      - run: ../configure
-        working-directory: build
-      - run: make dist
-        working-directory: build
+          ./autogen.sh
+          mkdir -p build
+          cd build
+          ../configure
+          make dist
       - name: Extract
         run: tar -xzvf thapi-*
         working-directory: build
-      - name: Configure
+      - name: Make Check
         run: |
-          cd `find . -type d -name "thapi-*"`
+          cd $(find . -type d -name "thapi-*")
           mkdir -p build
           cd build
           ../configure
-        working-directory: build
-      - name: Check
-        run: |
-          cd `find . -type d -name "thapi-*"`
-          cd build
```

Comment:
I don't like these, this goes contradictory to the philosophy of github actions and increasing the diff for no good reason. I think that if we want to do that we should do so in a different PR.